core/src/Streamly/Internal/Data/Stream/Top.hs

```diff
@@ -206,129 +253,196 @@ filterStreamWith :: Monad m =>
 filterStreamWith fld member s1 s2 =
     Stream.concatEffect
         $ do
-            xs <- Stream.fold fld s1
-            return $ Stream.filter (`member` xs) s2
+            xs <- Stream.fold fld s2
+            return $ Stream.filter (`member` xs) s1
```

Comment:
Is this quadratic?
Can we use a set here?

Reply:
This is for unconstrained version. The version with Set would be separate with an Ord constraint.